infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/SnapshotWriter.java

```diff
@@ -0,0 +1,16 @@
+package org.corfudb.infrastructure.logreplication.receive;
+
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationEntry;
+
+import java.util.List;
+
+/**
+ * The snapshot full sync engine will call the snapshot writer api apply to apply messages it has received.
+ * It will guarantee the ordering of the message that pass to the snapshot writer.
+ */
+public interface SnapshotWriter {
```

Comment:
Do owe further need this interface? It's always the same implementation of snapshot writer, right?